### PR TITLE
EOS-25125 Fetch local paths from confstore

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
Modified code to to fetch paths for consul and hax from confstore
Also modified code to get consul endpoints from confstore
Destroying leader session after consul start
Copying 'consul-server-conf.json' to '{hare_local_dir}/consul/config'' directory
Updating elect-rc-leader watcher entry in consul-server-conf.json to take config-dir as input

Signed-off-by: Swapnil Gaonkar <swapnil.gaonkar@seagate.com>

### CODE REVIEW

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -191,11 +194,29 @@ def _start_consul(utils: Utils, stop_event: Event, hare_local_dir: str):
...
                                    log_dir=log_dir, data_dir=data_dir,
                                    config_dir=config_dir, peers=peers)
     consul_starter.start()
+
+    try:
+        util: ConsulUtil = ConsulUtil()
+        sess = util.get_leader_session_no_wait()
+        util.destroy_session(sess)
+    except Exception:
+        logging.info('No leader is selected yet')
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:218]: s/selected/elected

File: provisioning/miniprov/hare_mp/main.py [hunk 2]
```diff
@@ -254,8 +275,9 @@ def prepare(args):
     url = args.config[0]
     utils = Utils(ConfStoreProvider(url))
     stop_event = Event()
-    local_dir = '/var/hare'
-    consul_starter = _start_consul(utils, stop_event, local_dir)
+    conf_dir = get_config_dir(url)
+    local_dir = f'{conf_dir}'
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:279]: why? what is the type of conf_dir? should be str, right?

File: provisioning/miniprov/hare_mp/main.py [hunk 3]
```diff
@@ -329,12 +351,15 @@ def start_hax_with_systemd():
 #         raise RuntimeError(f'Error while initializing cluster :key={error}')
 
 
-def start_hax_and_consul_without_systemd(config_url: str, utils: Utils):
-    local_dir = '/var/hare'
+def start_hax_and_consul_without_systemd(config_url: str,
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:354]: config_url not used.

File: provisioning/miniprov/hare_mp/main.py (hunk 1 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py]: Done

File: provisioning/miniprov/hare_mp/main.py (hunk 2 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py]: Done

File: provisioning/miniprov/hare_mp/main.py (hunk 3 above)
Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py]: Done

**Comment by SwapnilGaonkar7** [comment]
@mssawant Addressed review comments

File: provisioning/miniprov/hare_mp/main.py [hunk 4]
```diff
@@ -181,7 +181,10 @@ def unsupported_feature(url: str):
         logging.error('Error reporting hare unsupported features (%s)', error)
 
 
-def _start_consul(utils: Utils, stop_event: Event, hare_local_dir: str):
+def _start_consul(utils: Utils,
+                  stop_event: Event,
+                  hare_local_dir: str,
+                  url: str):
     log_dir = f'{hare_local_dir}/consul/log'
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:188]: Let's create these directories in a separate function, may be `create_consul_env()` and invoke it in prepare stage before invoking _start_consul().

Comment (SwapnilGaonkar7) [reply provisioning/miniprov/hare_mp/main.py:185]: Done

**Comment by mssawant** [comment]
@SwapnilGaonkar7, looks good, as discussed the only thing remains is updating elect-rc-leader watcher entry in consul-server-conf.json. Once done, can you test and post the the results please.

**Review by suvratjoshi — APPROVED** [review]

**Pull request merged**